elements-harness/src/elementd_rpc.rs

```diff
@@ -1,65 +1,121 @@
+    async fn getnewaddress(&self) -> Address;
+    async fn generatetoaddress(&self, blocks: u8, address: String) -> Vec<String>;
+    async fn sendtoaddress(&self, address: String, amount: f64) -> Txid;
+    async fn getaddressinfo(&self, address: String) -> AddressInfo;
+    async fn getrawtransaction(&self, tx_id: Txid) -> String;
+}
 
-#[derive(Debug, Clone)]
+#[jsonrpc_client::implement_async(ElementsRpc)]
+#[derive(Debug)]
 pub struct Client {
-    rpc_client: json_rpc::Client,
+    inner: reqwest::Client,
+    base_url: reqwest::Url,
 }
 
 impl Client {
-    pub fn new(url: Url) -> Self {
-        Client {
-            rpc_client: json_rpc::Client::new(url),
-        }
-    }
-
-    pub async fn network(&self) -> Result<String> {
-        let blockchain_info = self.blockchain_info().await?;
-
-        Ok(blockchain_info.chain)
-    }
-
-    async fn blockchain_info(&self) -> Result<BlockchainInfo> {
-        let blockchain_info = self
-            .rpc_client
-            .send::<Vec<()>, BlockchainInfo>(json_rpc::Request::new(
-                "getblockchaininfo",
-                vec![],
-                JSONRPC_VERSION.into(),
-            ))
-            .await?;
-
-        Ok(blockchain_info)
+    pub fn new(base_url: String) -> Result<Self> {
+        Ok(Self {
+            inner: reqwest::Client::new(),
+            base_url: base_url.parse()?,
+        })
     }
 }
 
 #[derive(Debug, Deserialize)]
 struct BlockchainInfo {
-    chain: String,
+    pub chain: String,
     mediantime: u32,
 }
 
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+pub struct AddressInfo {
+    pub address: Address,
+    #[serde(rename = "scriptPubKey")]
+    pub script_pub_key: Script,
+    #[serde(rename = "ismine")]
+    pub is_mine: bool,
+    pub solvable: bool,
+    #[serde(rename = "iswatchonly")]
+    pub is_watch_only: bool,
+    #[serde(rename = "isscript")]
+    pub is_script: bool,
+    #[serde(rename = "iswitness")]
+    pub is_witness: bool,
+    pub witness_version: u64,
+    pub witness_program: String,
+    #[serde(rename = "ischange")]
+    pub is_change: bool,
+    pub labels: Vec<String>,
+}
+
 #[cfg(all(test))]
 mod test {
     use super::*;
     use crate::Elementsd;
+    use std::str::FromStr;
     use testcontainers::clients;
 
     #[tokio::test]
     async fn get_network_info() {
         let tc_client = clients::Cli::default();
         let (client, _container) = {
             let blockchain = Elementsd::new(&tc_client, "0.18.1.9").unwrap();
+            (
+                Client::new(blockchain.node_url.clone().into_string()).unwrap(),
+                blockchain,
+            )
+        };
 
-            (Client::new(blockchain.node_url.clone()), blockchain)
+        let blockchain_info: BlockchainInfo = client.getblockchaininfo().await.unwrap();
+        let network = blockchain_info.chain;
+
+        assert_eq!(network, "elementsregtest")
+    }
+
+    #[tokio::test]
+    async fn rpc_sanity_tests() {
```

Comment:
:thinking: I'm not a big fan of tests that test too much, but I guess you want to avoid spinning up a container for every test. I would rather have individual tests and take the test speed hit though.

Reply:
Me neither. I'm not sure if we should keep this test at all... I rather wanted to play around with the library. 

Shall I remove the test and we only add tests for functions as needed?

Reply:
That sounds good!

Reply:
... remove ...